util/file_reader_writer.cc

```diff
@@ -502,7 +502,8 @@ class ReadaheadRandomAccessFile : public RandomAccessFile {
     if (prefetch_offset == buffer_offset_) {
       return Status::OK();
     }
-    return ReadIntoBuffer(prefetch_offset, offset - prefetch_offset + n);
+    return ReadIntoBuffer(prefetch_offset,
+                          Roundup(offset + n, alignment_) - prefetch_offset);
```

Comment:
@lightmark we have daily tests run on non-tmpfs and some Travis doesn't run on tmpfs. We do have platform to catch issues like this. Also, what Karthik did previously was to hack the code in unit tests, so that it in some tests it go through the whole direct I/O code path, except that in the end file descriptors are opened without O_DIRECT. In this way all those asserts can be checked to see whether alignment is guaranteed. I think you've done something like that in unit test EnvPosixTestWithParam.Preallocation. You can apply the same trick and cover DB end-to-end.

Next time if you find tests are hard to write or run, please talk to people for ideas. It's not an excuse of skipping testing.